### PR TITLE
python312Packages.webassets: drop nose dependency

### PR DESCRIPTION
**Pull request by Sigmanificient**
## Description of changes

- #326513

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/flask-assets/default.nix [hunk 1]
```diff
@@ -13,13 +14,21 @@ buildPythonPackage rec {
...
+    (fetchpatch2 {
+      url = "https://github.com/miracle2k/flask-assets/commit/56e06dbb160c165e0289ac97496354786fe3f3fd.patch?full_index=1";
+      hash = "sha256-Feo7gHHmHtWRB+3XvlECdU4i5rpyjyKEYEUCuy24rf4=";
+    })
+  ];
+
+  postPatch = ''
     substituteInPlace tests/test_integration.py --replace 'static_path=' 'static_url_path='
     substituteInPlace tests/test_integration.py --replace "static_folder = '/'" "static_folder = '/x'"
     substituteInPlace tests/test_integration.py --replace "'/foo'" "'/x/foo'"
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/flask-assets/default.nix:34]:
```suggestion
    substituteInPlace tests/test_integration.py \
      --replace-fail 'static_path=' 'static_url_path=' \
      --replace-fail "static_folder = '/'" "static_folder = '/x'" \
      --replace-fail "'/foo'" "'/x/foo'"
```

File: pkgs/development/python-modules/webassets/default.nix [hunk 2]
```diff
@@ -2,40 +2,66 @@
...
-  doCheck = false;
-  checkPhase = "py.test";
+  postPatch = ''
+    # Fix thread attribute "isAlive"
+    substituteInPlace tests/test_script.py \
+      --replace-fail "isAlive" "is_alive"
+  '';
+
+  disabledTestPaths = [
+    "tests/test_filters.py" # no module "distutils"
```
[the remark this answers is not in the document]
Comment (Sigmanificient) [reply pkgs/development/python-modules/webassets/default.nix:57]: in `nativeCheckInputs`?

[the remark this answers is not in the document]
Comment (Sigmanificient) [reply pkgs/development/python-modules/webassets/default.nix:57]: Still the same outcome:
```
____________________ ERROR collecting tests/test_filters.py ____________________
ImportError while importing test module '/build/webassets-2.0/tests/test_filters.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/nix/store/l014xp1qxdl6gim3zc0jv3mpxhbp346s-python3-3.12.4/lib/python3.12/importlib/__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/test_filters.py:15: in <module>
    from distutils.spawn import find_executable
E   ModuleNotFoundError: No module named 'distutils'
```

**Review by SuperSandro2000 — APPROVED** [review]

File: pkgs/development/python-modules/webassets/default.nix (hunk 2 above)
[the remark this answers is not in the document]
Comment (emilazy) [reply pkgs/development/python-modules/webassets/default.nix:57]: Try adding `distutils` to `dependencies` or `nativeCheckInputs` as appropriate.

**Review by emilazy — CHANGES_REQUESTED** [review]
Result of `nixpkgs-review pr 334483` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>12 packages failed to build:</summary>
  <ul>
    <li>octoprint</li>
    <li>octoprint.dist</li>
    <li>pyload-ng</li>
    <li>pyload-ng.dist</li>
    <li>python311Packages.dukpy</li>
    <li>python311Packages.dukpy.dist</li>
    <li>python311Packages.flask-assets</li>
    <li>python311Packages.flask-assets.dist</li>
    <li>python312Packages.dukpy</li>
    <li>python312Packages.dukpy.dist</li>
    <li>python312Packages.flask-assets</li>
    <li>python312Packages.flask-assets.dist</li>
  </ul>
</details>
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.webassets</li>
    <li>python311Packages.webassets.dist</li>
    <li>python312Packages.webassets</li>
    <li>python312Packages.webassets.dist</li>
  </ul>
</details>

```
substituteStream() in derivation python3.12-flask-assets-2.1.0: ERROR: pattern static_path= doesn't match anything in file 'tests/test_integration.py'
```

**Comment by emilazy** [comment]
@pyrox0 Did you want to handle this one too? Looks like the existing substitutions need updating or dropping.

**Comment by pyrox0** [comment]
Submitted #348621, which builds on top of this PR and solves outstanding review comments.

**Pull request closed without merging**